xonsh/foreign_shells.py

```diff
@@ -243,9 +244,10 @@ def foreign_shell_data(shell, interactive=True, login=False, envcmd=None,
     elif currenv is not None:
         currenv = dict(currenv)
     try:
+        # start new session to avoid hangs (doesn't work on Cygwin though)
+        newsess = not (sys.platform == "cygwin")
```

Comment:
Would you mind making an `ON_CYGWIN` variable in `xonsh.platform`, please, importing and and using this instead? I think it is a little more general.